cooltools/cli/compute_saddle.py

```diff
@@ -0,0 +1,515 @@
+
+
+    # read expected and make preparations for validation,
+    # it's contact_type dependent:
+    if contact_type == "cis":
+        # that's what we expect as column names:
+        expected_columns = ['chrom', 'diag', 'n_valid', expected_name]
+        # what would become a MultiIndex:
+        expected_index = ['chrom', 'diag']
+        # expected dtype as a rudimentary form of validation:
+        expected_dtype = {'chrom':np.str, 'diag':np.int64, 'n_valid':np.int64, expected_name:np.float64}
+        # unique list of chroms mentioned in expected_path:
+        get_exp_chroms = lambda df: df.index.get_level_values("chrom").unique()
+        # compute # of bins by comparing matching indexes:
+        get_exp_bins   = lambda df,ref_chroms,_: df.index.get_level_values("chrom").isin(ref_chroms).sum()
+    elif contact_type == "trans":
+        # that's what we expect as column names:
+        expected_columns = ['chrom1', 'chrom2', 'n_valid', expected_name]
+        # what would become a MultiIndex:
+        expected_index = ['chrom1', 'chrom2']
+        # expected dtype as a rudimentary form of validation:
+        expected_dtype = {'chrom1':np.str, 'chrom2':np.str, 'n_valid':np.int64, expected_name:np.float64}
+        # unique list of chroms mentioned in expected_path:
+        get_exp_chroms = lambda df: np.union1d(df.index.get_level_values("chrom1").unique(),
+                                           df.index.get_level_values("chrom2").unique())
+        # no way to get bins from trans-expected, so just get the number:
+        get_exp_bins   = lambda _1,_2,correct_bins: correct_bins
+    else:
+        raise ValueError("Incorrect contact_type: {}, ".format(contact_type),
+            "Should have been caught by click.")
+    # use 'usecols' as a rudimentary form of validation,
+    # and dtype. Keep 'comment' and 'verbose' - explicit,
+    # as we may use them later:
+    expected = pd.read_table(
+                            expected_path,
+                            usecols  = expected_columns,
+                            index_col= expected_index,
+                            dtype    = expected_dtype,
+                            comment  = None,
+                            verbose  = False)
+
+
+    # read BED-file :
+    track_columns = ['chrom', 'start', 'end', track_name]
+    # specify dtype as a rudimentary form of validation:
+    track_dtype = {'chrom':np.str, 'start':np.int64, 'end':np.int64, track_name:np.float64}
+    track = pd.read_table(
+                    track_path,
+                    usecols  = [0,1,2,3],
+                    names    = track_columns,
+                    dtype    = track_dtype,
+                    comment  = None,
+                    verbose  = False)
+    # # should we use bioframe as an alternative?:
+    # # ...
+    # from bioframe.io import formats
+    # from bioframe.schemas import SCHEMAS
+    # formats.read_table(track_path, schema=SCHEMAS["bed4"])
```

Comment:
Technically, these types of files with non-overlapping segmentations of quantitative data are called [bedGraph](http://genome.ucsc.edu/goldenPath/help/bedgraph.html), so it would be `schema='bedGraph'`. But bioframe isn't ready to be a dependency yet, though soon it should.

Reply:
mentioned that we might switch to `bioframe` in the future for input validation etc.
corrected schema reference.